src/node_buffer.cc

```diff
@@ -1190,6 +1190,40 @@ void Swap64(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(args[0]);
 }
 
+bool FastIsUtf8(v8::Local<v8::Value>,
+                Local<Value> buffer,
+                // NOLINTNEXTLINE(runtime/references) This is V8 api.
+                FastApiCallbackOptions& options) {
+  TRACK_V8_FAST_API_CALL("buffer.isUtf8");
+  ArrayBufferViewContents<uint8_t> view(buffer);
+  if (view.WasDetached()) {
```

Comment:
Is this necessary? A detached view really ought to be handled no differently than a zero-length view.

Reply:
This follows the same pattern of slow path

Reply:
Yeah, I get that... just looking at it now I'm not convinced throwing is the right thing.